packages/node_modules/pouchdb-core/src/setup.js

```diff
@@ -127,4 +128,60 @@ PouchDB.fetch = function (url, opts) {
   return fetch(url, opts);
 };
 
+function ActiveTasks() {
+  this.tasks = {};
+  this.enabled = true;
+}
+
+ActiveTasks.prototype.list = function () {
+  return this.tasks;
+};
+
+ActiveTasks.prototype.add = function (task) {
+  // substr just for testing
+  const id = uuidv4().substr(0,8);
+  this.tasks[id] = {
+    id,
+    name: task.name,
+    total_items: task.total_items,
+    created_at: new Date().toJSON()
+  };
+  return id;
+};
+
+ActiveTasks.prototype.get = function (id) {
+  return this.tasks[id];
+};
+
+ActiveTasks.prototype.remove = function (id) {
+  delete this.tasks[id];
+  return this.tasks;
+};
+
+ActiveTasks.prototype.update = function (id, updatedTask) {
+  const task = this.tasks[id];
+  if (typeof task !== 'undefined') {
+    const mergedTask = {
+      id: task.id,
+      name: task.name,
+      created_at: task.created_at,
+      total_items: updatedTask.total_items || task.total_items,
+      completed_items: updatedTask.completed_items || task.completed_items,
+      updated_at: new Date().toJSON()
+    };
+    this.tasks[id] = mergedTask;
+  }
+  return this.tasks;
+};
+
+ActiveTasks.prototype.enable = function () {
+  this.enabled = true;
+};
+
+ActiveTasks.prototype.disable = function () {
+  this.enabled = false;
+};
```

Comment:
we can quite likely remove the opt-in switch since the computational impact of this feature is negligible.